From af9a42926fc8c86b95153181d554acb1e893b3f5 Mon Sep 17 00:00:00 2001
From: Tomas Henzl <thenzl@redhat.com>
Date: Mon, 8 Sep 2014 14:32:12 -0400
Subject: [scsi] hpsa: fix bad -ENOMEM return value in hpsa_big_passthru_ioctl

Message-id: <1410186734-19986-82-git-send-email-thenzl@redhat.com>
Patchwork-id: 92085
O-Subject: [RHEL6.6 e-stor PATCH 81/83] hpsa: fix bad -ENOMEM return value in hpsa_big_passthru_ioctl
Bugzilla: 1069185
RH-Acked-by: Maurizio Lombardi <mlombard@redhat.com>
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Ewan Milne <emilne@redhat.com>

When copy_from_user fails, return -EFAULT, not -ENOMEM

Signed-off-by: Stephen M. Cameron <scameron@beardog.cce.hp.com>
Reported-by: Robert Elliott <elliott@hp.com>
Reviewed-by: Joe Handzik <joseph.t.handzik@hp.com>
Reviewed-by: Scott Teel <scott.teel@hp.com>
Reviewed by: Mike MIller <michael.miller@canonical.com>
Cc: <stable@vger.kernel.org>
Signed-off-by: Christoph Hellwig <hch@lst.de>

(cherry picked from commit 0758f4f732b08b6ef07f2e5f735655cf69fea477)
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/hpsa.c b/drivers/scsi/hpsa.c
index b131813..de21a90 100644
--- a/drivers/scsi/hpsa.c
+++ b/drivers/scsi/hpsa.c
@@ -5099,7 +5099,7 @@ static int hpsa_big_passthru_ioctl(struct ctlr_info *h, void __user *argp)
   }
   if (ioc->Request.Type.Direction & XFER_WRITE) {
    if (copy_from_user(buff[sg_used], data_ptr, sz)) {
-    status = -ENOMEM;
+    status = -EFAULT;
     goto cleanup1;
    }
   } else
-- 
1.7.1